### ISSUE TITLE
Source: prettierConfig doesn't work

### ISSUE THREAD

**Issue by weeebdev** [storybookjs/storybook#15532]
**Describe the bug**
When I try to configure source-loader it just doesn't work.

**System**
```bash
Environment Info:

  System:
    OS: Linux 5.4 Ubuntu 20.04.2 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
  Binaries:
    Node: 14.17.2 - ~/.nvm/versions/node/v14.17.2/bin/node
    npm: 6.14.13 - ~/.nvm/versions/node/v14.17.2/bin/npm
  Browsers:
    Chrome: 91.0.4472.114
    Firefox: 89.0.2
  npmPackages:
    @storybook/addon-actions: ^6.3.4 => 6.3.4 
    @storybook/addon-docs: ^6.3.4 => 6.3.4 
    @storybook/addon-essentials: ^6.3.4 => 6.3.4 
    @storybook/addon-links: ^6.3.4 => 6.3.4 
    @storybook/node-logger: ^6.3.4 => 6.3.4 
    @storybook/preset-create-react-app: ^3.2.0 => 3.2.0 
    @storybook/react: ^6.3.4 => 6.3.4 
```

**Additional context**
```js
module.exports = {
  webpackFinal: async (config, { configType }) => {
    config.module.rules.push({
      test: /\.stories\.tsx?$/,
      loaders: [
        {
          loader: require.resolve('@storybook/source-loader'),
          options: {
            parser: 'typescript',
            options: {
              prettierConfig: {
                printWidth: 100,
                singleQuote: true,
              },
            },
          },
        },
      ],
      enforce: 'pre',
    });

    return config;
  },
  stories: ['../src/**/*.stories.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
  addons: [
    '@storybook/addon-links',
    '@storybook/preset-create-react-app',
    {
      name: '@storybook/addon-docs',
      options: {
        sourceLoaderOptions: null,
      },
    },
    '@storybook/addon-essentials',
  ],
};
```

**Comment by sag1v** [comment]
Happens to me as well on v7.0.17

**Comment by sag1v** [comment]
@weeebdev did you find a solution?

**Comment by weeebdev** [comment]
> @weeebdev did you find a solution?

Honestly, I do not remember 😂

**Comment by chakAs3** [comment]
Hi @sag1v  actually for story source. we don't use anymore prettier,  now it is ReactSyntaxHighlighter so any config will be omitted. However we are thinking of going back to prettier if it is something you would like to have back

**Comment by sag1v** [comment]
Hi @chakAs3 So this means we can't pass `prettierConfig` anymore?
You might want to update the [DOCS](https://storybook.js.org/addons/@storybook/addon-storysource) if so.

My issue is that in v6 i got the source code pretty much the same way i had it in my stories, now i get a bit of an ugly version of it.

i'm getting the source code via the usage of the `useOf` hook:

```
const resolvedStory = useOf(SomeComponent);
const sourceCode = resolvedStory.story.parameters.docs.source.originalSource;
```

I'm doing some logic there and injecting missing import statements etc..
But when inspecting the `sourceCode` i see that i get an "un-pretty" version of it even before i change it.

Is there a way i could pass it through prettier on my own via an addon or a plugin or something? I can't use prettier in this code becuase this one runs on the browser, i need to hook on the build phase. Is this something i can do?

If not, i may try to add a webpack plugin of my own.

**Comment by chakAs3** [comment]
I will double-check, but we had already 1 issue with not wrapping long lines, and I'm sure we are not using prettier in Story source loader

**Comment by sag1v** [comment]
> I will double-check, but we had already 1 issue with not wrapping long lines, and I'm sure we are not using prettier in Story source loader 

Exactly my issue (mostly), long lines.

What was the reason to drop prettier for this version?

**Comment by chakAs3** [comment]
@sag1v i don't know why exactly. i have a fix for long lines already I'm fixing some issues related to the update, I will tag the PR to follow up with it

**Comment by sag1v** [comment]
> @sag1v i don't know why exactly. i have a fix for long lines already I'm fixing some issues related to the update, I will tag the PR to follow up with it 

Thanks 🙏
In my opinion the ideal solution would be to keep the source formatting as is. Do you think it's possible?

**Comment by chakAs3** [comment]
> > @sag1v i don't know why exactly. i have a fix for long lines already I'm fixing some issues related to the update, I will tag the PR to follow up with it
> 
> Thanks 🙏 In my opinion the ideal solution would be to keep the source formatting as is. Do you think it's possible?

Yes I think so,  then the user can format it as he likes, and it is the real source, he can use his own linter and prettier.

**Comment by sag1v** [comment]
> > > @sag1v i don't know why exactly. i have a fix for long lines already I'm fixing some issues related to the update, I will tag the PR to follow up with it
> > 
> > Thanks 🙏 In my opinion the ideal solution would be to keep the source formatting as is. Do you think it's possible?
> 
> Yes I think so,  then the user can format it as he likes, and it is the real source, he can use his own linter and prettier. 

Indeed. This is the reason I think it's ideal.
How do you think we can achieve that, i mean what makes the code formatting change now?

**Comment by sag1v** [comment]
@chakAs3 BTW, do you have a suggestion for how can i workaround this issue until it gets a proper fix? Can i somehow shim in the source loader process and pass the code via prettier? It has to be in build time, not runtime

**Comment by chakAs3** [comment]
@sag1v it is merged, you can check using  next branch

**Comment by shilman** [comment]
Shiver me timbers!! I just released https://github.com/storybookjs/storybook/releases/tag/v7.1.0-alpha.27 containing PR #22541 that references this issue. Upgrade today to the `@future` NPM tag to try it out!

```
npx sb@next upgrade --tag future
```

**Comment by mayank99** [comment]
after 7.1.0-alpha.27, the long lines _do_ wrap but they are not formatted (what we really need is line-breaks or preserve original formatting)
<img width="421" alt="image" src="https://github.com/storybookjs/storybook/assets/9084735/32cf6908-9c7a-44f5-baf7-25319ef33d7d">

i have a related issue currently open #22362

**Comment by sag1v** [comment]
@chakAs3 
I think we should figure the root cause for the change in the code format and handle that (prevent it somehow)

**Comment by sag1v** [comment]
For now my workaround is to use prettier in runtime (browser) via `prettier/standalone` and format the source code manually.
Hopefully this will get resolved soon.

**Comment by chakAs3** [comment]
> Also i think that this issue should be re-opened

Yes @sag1v I believe this is not the final fix, we already discuss this, I really don't like react-syntax-highlighter, which is really limited,  we will figure out a proper solution, at least the user can preserve his format.
